File: concepts/big-o/linear-time-example/java/LinearTimeExample.java
public class LinearTimeExample {
    public static void main(String[] args){
        int[] sampleArray = {3, 1, 4, 5, 9, 2, 6, 5};
        // 9
        System.out.println(findMax(sampleArray));
    }

    public static int findMax(int[] arr){
        int maxVal = arr[0];

        for (int i = 1; i < arr.length; i++){
            if (arr[i] > maxVal){
                maxVal = arr[i];
            }
        }

        return maxVal;
    }
}

File: concepts/big-o/linear-time-example/cpp/linearTimeExample.cpp
// g++ linearTimeExample.cpp -o linearTimeExample -std=c++20 && ./linearTimeExample

#include <iostream>
#include <vector>

int findMax(const std::vector<int> &vec)
{
    int maxVal{vec[0]};

    for (size_t i{0}; i < vec.size(); ++i)
    {
        if (vec[i] > maxVal)
        {
            maxVal = vec[i];
        }
    }

    return maxVal;
}

int main()
{
    std::vector<int> sampleVector{3, 1, 4, 5, 9, 2, 6, 5};

    // 9
    std::cout << findMax(sampleVector) << std::endl;

    return 0;
}